back/app/serializers/web_api/v1/phase_serializer.rb

```diff
@@ -11,37 +11,36 @@ class WebApi::V1::PhaseSerializer < WebApi::V1::BaseSerializer
     :reacting_dislike_enabled, :reacting_dislike_method, :reacting_dislike_limited_max,
     :allow_anonymous_participation, :presentation_mode, :ideas_order, :input_term
 
-  with_options if: proc { |phase|
-    phase.voting?
-  } do
-    attribute :voting_method
-    attribute :voting_max_total
-    attribute :voting_min_total
-    attribute :voting_max_votes_per_idea
-    attribute :baskets_count
-    attribute :voting_term_singular_multiloc do |phase|
-      phase.voting_term_singular_multiloc_with_fallback
-    end
-    attribute :voting_term_plural_multiloc do |phase|
-      phase.voting_term_plural_multiloc_with_fallback
-    end
-  end
-
-  with_options if: proc { |phase|
-    phase.native_survey?
-  } do
-    attribute :native_survey_title_multiloc
-    attribute :native_survey_button_multiloc
+  %i[
+    voting_method voting_max_total voting_min_total
+    voting_max_votes_per_idea baskets_count
+    native_survey_title_multiloc native_survey_button_multiloc
+  ].each do |attribute_name|
+    attribute attribute_name, if: proc { |phase|
+      Factory.instance.participation_method_for(phase).supports_serializing?(attribute_name)
```

Comment:
Now that `phase.method` is public, I wonder if it would be cleaner to do a full search replace of `Factory.instance.participation_method_for(phase)` to `phase.method`.

Actually, I think it could be interesting to always return the participation_method instance from lib, instead of a string for `participation_method`.

Reply:
Yes, I agree with that, would make it a lot more convenient, and it's so central to our model.